pkg/controller/job/job_controller.go

```diff
@@ -1150,6 +1162,17 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			jobCtx.finishedCondition = newFailedConditionForFailureTarget(jobCtx.finishedCondition, jm.clock.Now())
 		}
 	}
+	if isSuccessCriteriaMetCondition(jobCtx.finishedCondition) {
+		// Append the interim SuccessCriteriaMet condition to update the job status with before finalizers are removed.
+		if successCriteriaMetCondition(jobCtx.job) == nil {
+			jobCtx.job.Status.Conditions = append(jobCtx.job.Status.Conditions, *jobCtx.finishedCondition)
```

Comment:
Yes, that's right. Since I will refine this logic as I discussed in another thread, this logic might be changed a little bit.

Reply:
> Oui c'est vrai. Puisque je vais affiner cette logique comme je l'ai expliqué dans un autre fil de discussion, cette logique pourrait être un peu modifiée.

Reply:
> Yes, that's right, but Since I will refine this logic as I discussed in another thread, this logic might be changed a little bit.

NVM. I adapted your suggestions here. Thanks.

Reply:
Done.